stytch/api/sessions.py

```diff
@@ -1,132 +1,190 @@
+        return GetResponse.from_json(await resp.json())
 
     def authenticate(
         self,
         session_token: Optional[str] = None,
         session_jwt: Optional[str] = None,
         session_duration_minutes: Optional[int] = None,
         session_custom_claims: Optional[Dict[str, Any]] = None,
-    ):
-        data: Dict[str, Any] = {}
-
-        if session_token:
-            data["session_token"] = session_token
-        if session_jwt:
-            data["session_jwt"] = session_jwt
-        if session_duration_minutes:
-            data["session_duration_minutes"] = session_duration_minutes
-        if session_custom_claims:
-            data["session_custom_claims"] = session_custom_claims
-
-        return self._post(
-            "{0}/authenticate".format(self.sessions_url),
-            data=data,
-        )
+    ) -> AuthenticateResponse:
+        payload: Dict[str, Any] = {}
+
+        if session_token is not None:
+            payload["session_token"] = session_token
+        if session_jwt is not None:
+            payload["session_jwt"] = session_jwt
+        if session_duration_minutes is not None:
+            payload["session_duration_minutes"] = session_duration_minutes
+        if session_custom_claims is not None:
+            payload["session_custom_claims"] = session_custom_claims
 
+        url = self.api_base.route_with_sub_url(self.sub_url, "authenticate")
+
+        resp = self.sync_client.post(url, json=payload)
+        return AuthenticateResponse.from_json(resp.json())
+
+    async def authenticate_async(
+        self,
+        session_token: Optional[str] = None,
+        session_jwt: Optional[str] = None,
+        session_duration_minutes: Optional[int] = None,
+        session_custom_claims: Optional[Dict[str, Any]] = None,
+    ) -> AuthenticateResponse:
+        payload: Dict[str, Any] = {}
+
+        if session_token is not None:
+            payload["session_token"] = session_token
+        if session_jwt is not None:
+            payload["session_jwt"] = session_jwt
+        if session_duration_minutes is not None:
+            payload["session_duration_minutes"] = session_duration_minutes
+        if session_custom_claims is not None:
+            payload["session_custom_claims"] = session_custom_claims
+
+        url = self.api_base.route_with_sub_url(self.sub_url, "authenticate")
+
+        resp = await self.async_client.post(url, json=payload)
+        return AuthenticateResponse.from_json(await resp.json())
+
+    # MANUAL(authenticate_jwt)
     def authenticate_jwt(
         self,
         session_jwt: str,
-        *,
         max_token_age_seconds: Optional[int] = None,
         session_custom_claims: Optional[Dict[str, Any]] = None,
-    ):
-        """Parse a JWT and verify the signature, preferring local verification over remote.
+    ) -> AuthenticateResponse:
+        # TODO: Remember to write a test since this is manually generated
```

Comment:
Gentle reminder 🙂

Reply:
Good call.. yeah, I realized I was missing all the tests for the manual methods :( I thought I brought those in already, but alas.

Reply:
Added tests. New coverage:
```
Name                                           Stmts   Miss  Cover
------------------------------------------------------------------
stytch/api/magic_links_email.py                  137    117    15%
stytch/api/passwords_email.py                     84     69    18%
stytch/api/webauthn.py                            76     57    25%
stytch/api/passwords_session.py                   30     17    43%
stytch/api/otp_whatsapp.py                        68     32    53%
stytch/api/magic_links.py                         70     30    57%
stytch/api/users.py                              221     80    64%
stytch/api/otp_email.py                           68     18    74%
stytch/api/otp_sms.py                             68     18    74%
stytch/api/oauth.py                               42     10    76%
stytch/api/sessions.py                           110     26    76%
stytch/api/crypto_wallets.py                      60     14    77%
stytch/api/otp.py                                 52     12    77%
stytch/api/passwords.py                          102     22    78%
stytch/api/passwords_existing_password.py         38      8    79%
stytch/api/totps.py                               88     18    80%
codegen/types/api.py                              72      6    92%
codegen/__init__.py                                0      0   100%
codegen/core/__init__.py                           0      0   100%
codegen/core/snippet_set.py                       67      0   100%
codegen/types/__init__.py                          0      0   100%
codegen/types/argument.py                         35      0   100%
codegen/types/http_method.py                      21      0   100%
codegen/types/method.py                           59      0   100%
codegen/types/response_type.py                    20      0   100%
codegen/types/templates/__init__.py                5      0   100%
stytch/__init__.py                                 3      0   100%
stytch/api/__init__.py                             0      0   100%
stytch/client.py                                  36      0   100%
stytch/core/__init__.py                            0      0   100%
stytch/core/api_base.py                           10      0   100%
stytch/core/http/client.py                        37      0   100%
stytch/core/models.py                            119      0   100%
stytch/models/__init__.py                          0      0   100%
stytch/models/crypto_wallets.py                   12      0   100%
stytch/models/magic_links.py                      17      0   100%
stytch/models/magic_links_email.py                13      0   100%
stytch/models/oauth.py                            13      0   100%
stytch/models/otp.py                              10      0   100%
stytch/models/otp_email.py                         8      0   100%
stytch/models/otp_sms.py                           8      0   100%
stytch/models/otp_whatsapp.py                      8      0   100%
stytch/models/passwords.py                        20      0   100%
stytch/models/passwords_email.py                   6      0   100%
stytch/models/passwords_existing_password.py       3      0   100%
stytch/models/passwords_session.py                 5      0   100%
stytch/models/sessions.py                         10      0   100%
stytch/models/totps.py                            26      0   100%
stytch/models/users.py                            50      0   100%
stytch/models/webauthn.py                         18      0   100%
stytch/version.py                                  1      0   100%
------------------------------------------------------------------
TOTAL                                           2026    554    73%
```